### ISSUE TITLE
Office.context.mailbox.item.sessionData.setAsync is undefined

### ISSUE THREAD

**Issue by barclayadam** [OfficeDev/office-js#4169]
# Provide required information needed to triage your issue

`Office.context.mailbox.item.sessionData.setAsync` is `undefined` on supported Outlook versions

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: PC
* Host [Excel, Word, PowerPoint, etc.]: Office
* Office version number: 16.0.17328.20108 / 16.0.17231.20236
* Operating System: Windows
 
## Expected behavior

`Office.context.mailbox.item.sessionData.setAsync` to exist and work. We use `sessionData` but have a fallback in case it is not available. We have various checks in place to make sure it works, because we cannot rely on the requirement set checking (proven by this bug again, which would have manifested if we were only checking the requirement set):

````typescript
function sessionDataSupported() {
    const ctx = context();
    const diags = diagnostics();

    if (ctx.mailbox.item.sessionData == undefined) {
        return false;
    }

    if (isMobile()) {
        // Although sessionData property exists, trying to save will result in Sys.FeatureNotEnabled error
        return false;
    }

    // Based on comment at https://github.com/OfficeDev/office-js/issues/2300#issuecomment-1028878708 SessionData is only properly supported on
    // PC when the version is > 16.0.14701.20290

    // Only check version number on Windows platform
    if (diags.platform !== Office.PlatformType.PC) {
        return true;
    }

    const versionAsNumber = parseInt(diags.version.replace(/\./g, ''));

    return versionAsNumber >= 1601470120290;
}
````

## Current behavior
"sessionData.setAsync is not a function" error is thrown

As a workaround to this issue I've added an additional check, as below:

```` typescript
    if (ctx.mailbox.item.sessionData.setAsync == undefined) {
        return false;
    }
````

From the first snippet you can also see I've had to do an explicit check for mobile, because `sessionData` as an object exists, but using any methods just throws. This object is generally problematic and not well defined.

## Steps to reproduce

1. Attempt to use `sessionData.setAsync` on an affected Outlook installation

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by manuelsidler** [comment]
@barclayadam the issue is already addressed #4154

**Comment by zhngx1** [comment]
Thanks for reporting this, it is a duplicate of #4154
